systemd/ceph-osd@.service.in

```diff
@@ -1,8 +1,10 @@
 [Unit]
 Description=Ceph object storage daemon osd.%i
-After=network-online.target local-fs.target time-sync.target ceph-mon.target
-Wants=network-online.target local-fs.target time-sync.target
 PartOf=ceph-osd.target
+After=network-online.target local-fs.target time-sync.target
+Before=ceph-osd.target
```

Comment:
Does this make sense? Why would this be both `PartOf` and `Before` `ceph-osd.target`?

Reply:
https://www.freedesktop.org/software/systemd/man/systemd.unit.html#PartOf=
> Configures dependencies similar to Requires=, but limited to stopping and restarting of units. When systemd stops or restarts the units listed here, the action is propagated to this unit. Note that this is a one-way dependency — changes to this unit do not affect the listed units.

https://www.freedesktop.org/software/systemd/man/systemd.unit.html#Before=
> Those two settings configure ordering dependencies between units. If unit foo.service contains the setting Before=bar.service and both units are being started, bar.service's start-up is delayed until foo.service has finished starting up.

Therefore specify both `PartOf` and `Before` could ensure **ALL** `ceph-osd@*.service` are stopped before `ceph-osd.target` get stopped.

As shown in https://github.com/ceph/ceph/pull/36776#issuecomment-684764395, the final result order looks good enough without conflict ;-)